Clarify elsewhen initial (#2647)
modelica/ModelicaSpecification: chapters/equations.tex

(such as simulation or linearization).

The equations of a when-clause are active during initialization, if and only if they are explicitly enabled with \lstinline!initial()!, and only in one of the two forms

> \lstinline!when initial() then! or \lstinline!when {$\ldots$, initial(), $\ldots$} then! (and similarly for \lstinline!elsewhen!  and algorithms see below).  In this case, the when-clause equations remain active during the

Reviewer: There is double space after `elsewhen`:
```suggestion
\lstinline!when initial() then! or \lstinline!when {$\ldots$, initial(), $\ldots$} then! (and similarly for \lstinline!elsewhen! and algorithms see below).  In this case, the when-clause equations remain active during the
```

> \lstinline!when initial() then! or \lstinline!when {$\ldots$, initial(), $\ldots$} then! (and similarly for \lstinline!elsewhen!  and algorithms see below).  In this case, the when-clause equations remain active during the

Reviewer: Perhaps _when-statements_ would be more to the point, and the _see below_ probably only refers to these?
```suggestion
(where \lstinline!when! may be replaced by \lstinline!elsewhen!) \lstinline!when initial() then! or \lstinline!when {$\ldots$, initial(), $\ldots$} then! (and similarly for when-statements, see below).  In this case, the when-clause equations remain active during the
```

Author: I first considered a text like that, but I didn't like it since there is a slight difference between elsewhen and when, in that when is always first - and I wanted to delay those obscure cases. The elsewhen-part below handles both.

Reviewer: To me it just seems unfortunate that `when`/`elsewhen` causes a lack of symmetry that is only syntactical.  This is why I ended up with the formulation suggested below (_if and only if they are the first clause…_) that captures the symmetry of the semantics.

> \lstinline!when initial() then! or \lstinline!when {$\ldots$, initial(), $\ldots$} then!. In this case, the algorithmic statements within the when-statement remain active during the whole

Reviewer: I suppose there should be an `elsewhen` clarification here as well, preserving the symmetry with the when-equations?

initialization phase.

An active when-clause inactivates the following \lstinline!elsewhen! (similarly as for when-clauses during simulation), but apart from that
the first \lstinline!elsewhen initial() then! or \lstinline!elsewhen {$\ldots$, initial(), $\ldots$} then! is similarly active during initialization as

> \lstinline!when initial() then! or \lstinline!when {$\ldots$, initial(), $\ldots$} then!.

Reviewer: Oh, maybe this was also included among the intended references of the _see below_ above?

I think we can say this more directly, if we go back to the first sentence:
> The equations of a when-clause are active during initialization, if and only if they are the first clause of the when-equation explicitly enabled with \lstinline!initial()!, …

\begin{nonnormative}

> That means that any subsequent \lstinline!elsewhen initial()! has no effect,
> similarly as \lstinline!when false then!.

Reviewer: Could we go as far as to speak of warnings?
```suggestion
A tool may warn in case an \lstinline!initial()! condition cannot activate its when-clause due to an earlier when-clause already being active during initialization.
```

Author: I considered it, but there are (as indicated) other similar cases as well, and to me it looks odd if we discuss warnings for the obscure cases - and not for the simple ones; and thought it sufficed to say that it has no effect.
